### PR TITLE
Upgrade to GitHub gitignore configuration

### PR DESCRIPTION
**Pull request by jpventura**
Upgrade `.gitignore` configuration in order to cover more IDE and editor automatically generated files.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,7 +1,53 @@
-.gradle
+# Built application files
+*.apk
+*.ap_
```
Comment (SUPERCILEX) [inline .gitignore]: The are covered by `build`.

File: .gitignore [hunk 2]
```diff
@@ -1,7 +1,53 @@
...
+
+# Files for the ART/Dalvik VM
+*.dex
```
Comment (SUPERCILEX) [inline .gitignore]: We don't use multi dexing

File: .gitignore [hunk 3]
```diff
@@ -1,7 +1,53 @@
...
+
+# Java class files
+*.class
```
Comment (SUPERCILEX) [inline .gitignore]: Also covered by `build`.

File: .gitignore [hunk 4]
```diff
@@ -1,7 +1,53 @@
...
+
+# Generated files
+bin/
+gen/
+out/
```
Comment (SUPERCILEX) [inline .gitignore]: gen is not used by Gradle and the others are covered by build I believe

File: .gitignore [hunk 5]
```diff
@@ -1,7 +1,53 @@
...
+
+# Gradle files
+.gradle/
+build/
```
Comment (SUPERCILEX) [inline .gitignore]: I believe this is wrong because you want to include sub directories.

File: .gitignore [hunk 6]
```diff
@@ -1,7 +1,53 @@
...
+
+# Local configuration file (sdk path, etc)
+local.properties
+
+# Proguard folder generated by Eclipse
+proguard/
```
Comment (SUPERCILEX) [inline .gitignore]: Eclipse is no longer supported.

File: .gitignore [hunk 7]
```diff
@@ -1,7 +1,53 @@
...
+
+# Log Files
+*.log
+
+# Android Studio Navigation editor temp files
+.navigation/
```
Comment (SUPERCILEX) [inline .gitignore]: I've never seen these before, etc.

File: .gitignore (hunk 3 above)
Comment (jpventura) [inline .gitignore]: +1

File: .gitignore (hunk 2 above)
Comment (jpventura) [inline .gitignore]: +1

File: .gitignore (hunk 1 above)
Comment (jpventura) [inline .gitignore]: +1

File: .gitignore (hunk 6 above)
Comment (jpventura) [inline .gitignore]: +1

File: .gitignore (hunk 4 above)
Comment (jpventura) [inline .gitignore]: This was used by Ant build system

**Comment by SUPERCILEX** [comment]
@jpventura sorry if I came off a little harsh, I was just trying to point out that your change added unessassary complexity. Thanks for your efforts though!

**Pull request closed without merging**